unittests/test-contracts/savanna/ibc/ibc.cpp

```diff
@@ -59,7 +52,7 @@ void svnn_ibc::_maybe_add_proven_root(const uint32_t block_num, const checksum25
 }
 
 template<typename Table>
-void svnn_ibc::_maybe_remove_from_cache(){
+void ibc::_maybe_remove_from_cache(){

```

Comment:
When removing at line 72, we should not remove all until `idx.end()` I think, since it wold remove all cache entries whether they are expired or not, I think we should go until `idx.upper_bound(now)`.

Also, we don't need the while loop, we can do 
```
auto end =  idx.upper_bound(now);
if (end == idx.end())
   end = last_itr;
idx.erase(idx.begin(), end)
``` 
I believe we can replace lines 65 to 76 with  this.

Reply:
Unfortunately, the CDT implementation of the multi_index interface only accepts a single iterator and will not work on a range, hence the reason why we need a loop here.

https://github.com/AntelopeIO/cdt/blob/main/libraries/eosiolib/contracts/eosio/multi_index.hpp#L1894